extensions/binding/org.eclipse.smarthome.binding.weatherunderground/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,459 @@
+				<options>
+					<option value="hPa">Hectopascal</option>
+					<option value="inHg">Inch of mercury</option>
+				</options>
+				<default>hPa</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="visibility" advanced="true">
+		<item-type>Number</item-type>
+		<label>Visibility</label>
+		<description>Visibility</description>
+		<state readOnly="true" pattern="%.1f" />
+		<config-description>
+			<parameter name="SourceUnit" type="text" required="true">
+				<label>Source Visibility Unit</label>
+				<description>Select the visibility unit provided by the Weather Underground service</description>
+				<options>
+					<option value="km">Kilometers</option>
+					<option value="mi">Miles</option>
+				</options>
+				<default>km</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="solarRadiation" advanced="true">
+		<item-type>Number</item-type>
+		<label>Solar Radiation</label>
+		<description>Solar radiation in W/m2</description>
+		<state readOnly="true" pattern="%.2f W/m2">
+		</state>
+	</channel-type>
+
+	<channel-type id="UVIndex" advanced="true">
+		<item-type>Number</item-type>
+		<label>UV Index</label>
+		<description>UV Index</description>
+		<state readOnly="true" pattern="%.1f">
+		</state>
+	</channel-type>
+
+	<channel-type id="rain">
+		<item-type>Number</item-type>
+		<label>Rain Fall Day</label>
+		<description>Rain fall during the day</description>
+		<category>Rain</category>
+		<state readOnly="true" pattern="%.2f" />
+		<config-description>
+			<parameter name="SourceUnit" type="text" required="true">
+				<label>Source Rain fall Unit</label>
+				<description>Select the rain fall unit provided by the Weather Underground service</description>
+				<options>
+					<option value="mm">Millimeters</option>
+					<option value="in">Inches</option>
+				</options>
+				<default>mm</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="snow">
+		<item-type>Number</item-type>
+		<label>Snow Fall</label>
+		<description>Snow fall</description>
+		<category>Rain</category>
+		<state readOnly="true" pattern="%.2f" />
+		<config-description>
+			<parameter name="SourceUnit" type="text" required="true">
+				<label>Source Snow fall Unit</label>
+				<description>Select the snow fall unit provided by the Weather Underground service</description>
+				<options>
+					<option value="cm">Centimeters</option>
+					<option value="in">Inches</option>
+				</options>
+				<default>cm</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="probaPrecipitation">
+		<item-type>Number</item-type>
+		<label>Precipitation Probability</label>
+		<description>Probability of precipitation in %</description>
+		<state readOnly="true" min="0" max="100" pattern="%d %%">
+		</state>
+	</channel-type>
+
+	<channel-type id="iconUrl" advanced="true">
```

Comment:
When implementing this binding, I think I have not yet added support for image element in Basic UI and Classic UI. It makes no sense at this time to use Image rather than URL.
The discussion relative to weather icons is still opened (in the weather general issue). Waiting for a conclusion, I just implemented an image.
I am now moving from string URL to Image.